smt/surrogate_models/krg_based.py

```diff
@@ -1006,7 +1010,18 @@ def _reduced_likelihood_function(self, theta):
         sigma2 = (rho**2.0).sum(axis=0) / (self.nt - p - q)
         reduced_likelihood_function_value = -(self.nt - p - q) * np.log10(
             sigma2.sum()
-        ) - self.nt * np.log10(detR)
+        ) - self.nt * np.log10(detR_noisy)
+
+        eval_noise = self.options["eval_noise"]
+        is_noise0 = self.options["noise0"] != [0.0]
+        # noisy KRG
+        sigma2_ri = None
+        is_noisy = eval_noise or is_noise0
+        sigma2_ri = None
+        if is_noisy:
```

Comment:
Prefer more compact:  `if self.options["eval_noise"] or self.options["noise0"] != [0.0]:   # is_noisy` instead of the five lines above and you can assign `par["sigma2_ri"]` directly